README.md

```diff
@@ -54,4 +54,26 @@ Create extension/myextension/design/standard/templates/esiwidget/widgets/usernam
 
 Use your new widget! In the relevant template, call the widget:
 
-    {mugo_esi_widget( 'username' )}
\ No newline at end of file
+    {mugo_esi_widget( 'username' )}
+
+You can pass parameters to the esi widget and they will be passed as a GET parameter, here is one example on how to pass the param 'tagID' to the widget:
+
+    {mugo_esi_widget( 'paramtest', hash( 'tagID', $tagID ) )}
+
+And you can read the parameter using the eZHTTPTool class:
+
+    <?php
+    class MyESIWidgets
+    {
+        public static function paramtestAction()
+        {
+            $tagID = eZHTTPTool::hasGetVariable('tagID') ? eZHTTPTool::getVariable('tagID') : false;
+            return array('tag_id' => $tagID);
+        }
+    }
+
+To avoid problems with Varnish cache, you can include the following code in the pagelayout:
```

Comment:
This is a must-have, so this should be re-worded:

"You must include the following code in your pagelayout in order to support ESI widgets in full view templates. This is to ensure that eZ Publish will pass the HTTP header informing Varnish that there are ESI tags in the response."